test/org/sagebionetworks/bridge/dynamodb/DynamoFPHSExternalIdentifierDaoTest.java

```diff
@@ -0,0 +1,133 @@
+import org.sagebionetworks.bridge.models.accounts.FPHSExternalIdentifier;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.google.common.collect.Lists;
+
+@ContextConfiguration("classpath:test-context.xml")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class DynamoFPHSExternalIdentifierDaoTest {
+
+    @Resource
+    private DynamoFPHSExternalIdentifierDao dao;
+    
+    @BeforeClass
+    public static void beforeClass() {
+        DynamoInitializer.init(DynamoFPHSExternalIdentifier.class);
+    }
+
+    @After
+    public void after() {
+        dao.deleteAll();
+    }
+    
+    @Test
+    public void verifyExternalIdDoesNotExist() throws Exception {
+        ExternalIdentifier externalId = new ExternalIdentifier("abc");
+        
+        assertFalse(dao.verifyExternalId(externalId));
+    }
+    
+    @Test
+    public void verifyExternalIdAlreadyRegistered() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("bar");
+        DynamoFPHSExternalIdentifier id2 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1, id2));
+        dao.registerExternalId(new ExternalIdentifier(id2.getExternalId()));
+        
+        assertTrue(dao.verifyExternalId(new ExternalIdentifier("bar")));
+        assertFalse(dao.verifyExternalId(new ExternalIdentifier("baz")));
+    }
+    
+    @Test(expected = EntityNotFoundException.class)
+    public void registerUnknownCodeThrowsException() throws Exception {
+        dao.registerExternalId(new ExternalIdentifier("bar"));
+    }
+    
+    @Test(expected = EntityAlreadyExistsException.class)
+    public void registerCodeAgainThrowsException() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1));
+        dao.registerExternalId(new ExternalIdentifier(id1.getExternalId()));
+        
+        dao.registerExternalId(new ExternalIdentifier("baz"));
+    }
+    
+    @Test
+    public void registerValidCodeOK() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1));
+        
+        dao.registerExternalId(new ExternalIdentifier("baz"));
+        
+        assertTrue(dao.getExternalIds().get(0).getRegistered());
+    }
+    
+    @Test
+    public void getExternalIds() throws Exception {
+        DynamoFPHSExternalIdentifier id1 = new DynamoFPHSExternalIdentifier("bar");
+        DynamoFPHSExternalIdentifier id2 = new DynamoFPHSExternalIdentifier("baz");
+        dao.addExternalIds(Lists.newArrayList(id1, id2));
+        dao.registerExternalId(new ExternalIdentifier(id1.getExternalId()));
+        
+        List<FPHSExternalIdentifier> identifiers = dao.getExternalIds();
+        assertEquals(2, identifiers.size());
```

Comment:
To make this less dependent on the initial state of the FPHSExternalId table, would it be better to check that identifiers.size() is >= 2, and check that the list contains "bar" (registered) and "baz" (not registered)?